linux/src/threadpool_linux.c

```diff
@@ -312,28 +246,17 @@ static int reallocate_threadpool_array(THREADPOOL* threadpool)
                 }
                 threadpool->task_array = temp_array;
 
-                // Ensure there are no gaps in the array
-                // Consume = 2
-                // Produce = 2
-                // [x x x x]
-                // Consume = 3
-                // Produce = 2
-                // [x x 0 x]
-                // During resize we will have to memmove in the gap at 2
-                // [x x 0 x 0 0 0 0]
-
                 int64_t insert_pos = interlocked_add_64(&threadpool->insert_idx, 0) % existing_count;
                 int64_t consume_pos = interlocked_add_64(&threadpool->consume_idx, 0) % existing_count;
-
                 uint32_t compress_count = 0;
+
+                /* Codes_SRS_THREADPOOL_LINUX_07_044: [ threadpool_schedule_work shall remove any gap in the task array. ]*/
```

Comment:
>all remove any gap [](http://example.com/codeflow?start=83&length=18)

I think a simpler way to do this is to memmove everything between the consume index and the size of the array before resize (so one element in the example above) to the end of the new resized array.

No iteration one by one, no gap filling, simply moving a bunch of entries.

Reply:
Let me know if it does not make sense and we can whiteboard it.

Reply:
done